spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -452,6 +456,22 @@ private void declareQueue(String beanName, Queue queue) {
 		addToAutoDeclareContext(beanName, queue);
 	}
 
+	private Exchange buildExchange(RabbitCommonProperties properties, String exchangeName) {
+		try {
+			// TODO Make the ctor public in Spring-AMQP - AMQP-695
+			Constructor<ExchangeBuilder> ctor = ExchangeBuilder.class.getDeclaredConstructor(String.class, String.class);
```

Comment:
Fine me this to merge and fix later.
Any way Spring AMQP is transitive dependency from Boot, which is transitive from Spring Cloud Build and so on.
So, we have to wait for an appropriate `starter` version anyway.
Perhaps a GH issue do not forget.

And yes, I agree about long-lived PRs. 
No need to go so far: the neighbor PR here in this project #22 is there for a long time and looks like it addresses a `routingKey` case as well 😄 
And yeah, right now that looks like obsolete and fully is superseded by this one. So, maybe it is a time to close that one ? 😉